contactmoment/src/test/kotlin/com/ritense/contactmoment/connector/ContactMomentConnectorIntTest.kt

```diff
@@ -0,0 +1,141 @@
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient
+
+@AutoConfigureWebTestClient(timeout = "36000")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class ContactMomentConnectorIntTest : BaseIntegrationTest() {
+
+    @Autowired
+    @Qualifier("contactMomentConnector")
+    lateinit var contactMomentConnector: Connector
+
+    @Autowired
+    lateinit var connectorService: ConnectorService
+
+    @Autowired
+    lateinit var connectorTypeInstanceRepository: ConnectorTypeInstanceRepository
+
+    @Autowired
+    lateinit var contactMomentProperties: ContactMomentProperties
+
+    @Autowired
+    lateinit var connectorDeploymentService: ConnectorDeploymentService
+
+    @Autowired
+    lateinit var connectorFluentBuilder: ConnectorFluentBuilder
+
+    lateinit var server: MockWebServer
+
+    @BeforeEach
+    internal fun setUp() {
+        startMockServer()
+        setupConnector()
+    }
+
+    @AfterEach
+    internal fun tearDown() {
+        server.shutdown()
+    }
+
+    @Test
+    fun `should create contactmoment`() {
+        val request = CreateContactMomentRequest(
+            "http://example.com",
+            "string",
+            "2019-08-24T14:15:22Z",
+            "string",
+            "string",
+            "str",
+            "string",
+            listOf("http://example.com"),
+            "gemeente",
+            "http://example.com",
+            CreateContactMomentRequest.MedewerkerIdentificatieRequest(
+                "string",
+                "string",
+                "string",
+                "string"
+            )
+        )
+
+        val contactMoment = (contactMomentConnector as ContactMomentConnector).createContactMoment(request)
+
+        Assertions.assertThat(contactMoment.url)
+            .isEqualTo("http://localhost:8006/contactmomenten/api/v1/contactmomenten/4e517f3f-1f2c-41cb-bf11-75c86c7ca51b")
+    }
+
+    fun startMockServer() {
+        val dispatcher: Dispatcher = object : Dispatcher() {
+            @Throws(InterruptedException::class)
+            override fun dispatch(request: RecordedRequest): MockResponse {
+                val response = when (request.path?.substringBefore('?')) {
+                    "/contactmomenten/api/v1/contactmomenten" -> mockResponseFromFile("/data/post-contact-moment.json")
+                    else -> MockResponse().setResponseCode(404)
+                }
+                return response
+            }
+        }
+        server = MockWebServer()
+        server.dispatcher = dispatcher
+        server.start()
+    }
+
+    private fun setupConnector() {
+        contactMomentProperties.url = server.url("/").toString()
+        contactMomentProperties.clientId = "valtimo-test"
+        contactMomentProperties.secret = " 41625e21-c4ef-487b-93fc-e46a25278d11"
+        connectorDeploymentService.deployAll(listOf(contactMomentConnector))
```

Comment:
Deze is niet nodig als je de connectorApplicationReadyEventListener niet mockt

Reply:
Helaas, als ik deze niet mock: 'connectorDeploymentService', dan gaat hij dingen deployen nog voor dat er database tabellen zijn en geeft hij een error. heb geen zin om uit te zoeken waarom